select2.ur

```diff
@@ -1,26 +1,80 @@
-type t = {Multi : bool,
-          Options : xml [Cselect, Body] [] [],
-          Selected : source (list string)}
+(* selector - describes the state of the select2 component.
+ *
+ * Parameterized by a type t corresponding to the underlying values, and by a
+ *   type -> type k corresponding to how t is wrapped - as a list or as an
+ *   an option, corresponding to multi and single selectors.
+ *
+ * This is necessary to get the functions to be nicely generic.
+ *)
+type selector k t =
+  {Options : xml [Cselect, Body] [] [],
+   Values : list t,
+   Selected : source (k int),
+   Multi : bool}
 
-fun create options =
-    s <- source [];
-    return {Multi = True, Options = options, Selected = s}
+(* Typeclasses describing how to unwrap the lists from the Select2 FFI into
+ * lists (in the case of multi) and options (in the case of single).
+ * 
+ * This is necessary to get the functions to be nicely generic.
+ *)
+con singleOrMulti k = list string -> k int
+fun singleOrMulti_option selections =
+  case selections of
+    (x :: _) => Some (readError x)
+  | [] => None
+fun singleOrMulti_list selections = List.mp readError selections
```

Comment:
I don't think we _need_ to abstract this or make it more generic, but if we were to, I think the we want something like [Haskell's `IsList`](https://hackage.haskell.org/package/base-4.19.1.0/docs/GHC-Exts.html#t:IsList).  Essentially, it's a type class with two functions: `toList : x -> list a` and `fromList : list a -> x`, such that `fromList (toList a) = a` (and notably, `toList (fromList a)` does not need to equal `a`).  This is easy to implement for both `list` and `option`, and then we can define `singleOrMulti` as requiring both `mappable` and `listLike` and being defined as `fn x => mp readError (toList x)`